packages/taco/src/config-validator.ts

```diff
@@ -0,0 +1,279 @@
+/**
+ * TACo Domain Configuration and Validation
+ *
+ * This module provides domain configuration management, validation utilities,
+ * and configuration processing for TACo operations across different networks.
+ */
+
+import { DomainName, DOMAINS, isViemClient } from '@nucypher/shared';
+import { ethers } from 'ethers';
+import type { PublicClient } from 'viem';
+
+import type { TacoClientConfig } from './client';
+
+/**
+ * Generic validation result interface
+ */
+export interface ValidationResult {
+  isValid: boolean;
+  errors: string[];
+}
+
+/**
+ * TACo Configuration Validator
+ *
+ * Validates TACo client configurations, domains, and provider compatibility.
+ * Provides both fast and full validation methods for TACo operations.
+ */
+export class TacoConfigValidator {
+  /**
+   * Get all supported TACo domain names
+   * @returns {DomainName[]} Array of supported TACo domain names ('lynx', 'tapir', 'mainnet')
+   */
+  static getSupportedDomains(): DomainName[] {
+    return Object.values(DOMAINS).map((domain) => domain.domain);
+  }
+
+  /**
+   * Check if domain is valid
+   * @param {DomainName} domain - TACo domain name to check ('lynx', 'tapir', 'mainnet')
+   * @returns {boolean} True if domain exists
+   */
+  static isValidDomain(domain: DomainName): boolean {
+    return !!domain && this.getSupportedDomains().includes(domain);
+  }
+
+  /**
+   * Get expected chain ID for domain from DOMAINS configuration
+   * @param {DomainName} domain - Domain name to look up
+   * @returns {number | undefined} Chain ID for the domain, undefined if not found
+   * @private
+   */
+  private static getExpectedChainId(domain: DomainName): number | undefined {
+    const domainEntry = Object.values(DOMAINS).find(
+      (domainConfig) => domainConfig.domain === domain,
+    );
+    return domainEntry?.chainId;
+  }
+
+  /**
+   * Validate ritual ID (basic validation - positive integer only)
+   * @param {number} ritualId - Ritual ID to validate
+   * @returns {boolean} True if valid (positive integer)
+   */
+  static isValidRitualId(ritualId: number): boolean {
+    return (
+      typeof ritualId === 'number' && Number.isInteger(ritualId) && ritualId > 0
+    );
+  }
+
+  /**
+   * Validate provider compatibility with domain
+   * @param {DomainName} domain - Domain name
+   * @param {PublicClient | ethers.providers.Provider} provider - Provider to validate (viem PublicClient or ethers Provider)
+   * @returns {Promise<boolean>} True if provider is valid for domain
+   */
+  static async isValidProvider(
+    domain: DomainName,
+    provider: PublicClient | ethers.providers.Provider,
```

Comment:
Why not use `ProviderLike` here?

Reply:
Sorry this was an oversight. Thanks for the catch.